### PR TITLE
feat: register with consul client

### PR DESCRIPTION
**Pull request by rfyiamcool**
### summary

1.  add `NewConsulRegisterWithClient` function. because app already consul client at startup, don't create client again.
2. foamt readme.md
3. I think it is necessary to mark in the document what to do if `health check` is not used. because some network architectures are complicate, consul cannot connect to the registered node. 😁  It will be more friendly to explain in the documentation.

### CODE REVIEW

**Comment by GuangmingLuo** [comment]
@rfyiamcool Hello，麻烦更新下go mod的 Kitex 版本，可以修复下 CI 问题

**Comment by GuangmingLuo** [comment]
@rfyiamcool Hello，rebase 下 master 代码

File: README.md [hunk 1]
```diff
@@ -143,7 +147,7 @@ func main() {
 
 ## Example
 
-See Server and Client in example/basic or example/custom-config.
+See Server and Client in [example/basic](https://github.com/rfyiamcool/registry-consul/tree/main/example/basic) or [example/custom-config](https://github.com/rfyiamcool/registry-consul/tree/main/example/custom-config).
```
Comment (baiyutang) [inline README.md]: the link should be `kitex-contrib/registry-consul`

Comment (baiyutang) [reply README.md]: @rfyiamcool  劳烦及时跟进下代码，过久的PR 可能会被关闭哈

Comment (rfyiamcool) [reply README.md]: done.

Comment (baiyutang) [reply README.md]: 这个链接要链接回上游源仓库 就是 kitex-contrib 的。 或者你可以用相对路径试试。

**Pull request merged**